app/models/state_file_base_intake.rb

```diff
@@ -83,12 +83,19 @@ def synchronize_filers_to_database
     update(attributes_to_update) if attributes_to_update.present?
   end
 
+  class SynchronizeError < StandardError; end
+
   def synchronize_df_dependents_to_database
     direct_file_data.dependents.each do |direct_file_dependent|
       dependent = dependents.find { |d| d.ssn == direct_file_dependent.ssn } || dependents.build
       dependent.assign_attributes(direct_file_dependent.attributes)
 
       dependent_json = direct_file_json_data.find_matching_json_dependent(dependent)
+
+      if direct_file_json_data.data.present? && dependent_json.nil?
+        raise SynchronizeError, "Could not find matching dependent #{dependent.id} with #{state_name} intake id: #{id}"
+      end
```

Comment:
should we also thrown an error if number of dependents on xml / json doesn't match? It's possible that xml has dependents that match some of the json dependents, but some that end up not having a json counterpart. Is that overkill?

Reply:
I think that's overkill, if we identify data integrity issues like that with IRS data during testing we can submit issues with them